instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/Network.kt

```diff
@@ -12,13 +22,62 @@ suspend fun BaseCoreNoCleanUpTest.withoutInternet(block: suspend () -> Unit) {
 }
 
 suspend fun withoutWifiAndMobileData(block: suspend () -> Unit) {
-    val uiAutomation = InstrumentationRegistry.getInstrumentation().uiAutomation
+    val reachability = ReachabilityFactory.reachability(null)
+    Log.d(LOG_TAG, "Got request to turn internet off, checking if it was present")
+    reachability.waitForNetworkStatus { it != NetworkStatus.NOT_REACHABLE }
```

Comment:
What does this line does? Are you waiting for the Internet to become available? Why?

Reply:
There are two reasons behind my motivation: 

1. In the rest of the logic I assume that the internet was present before turning it off. So decided to add an explicit check for that in the beginning
2. To make sure that the reachability service is warmed up and works as expected

Reply:
But then it should be checked at the beginning of the test?

Reply:
why? 🤔

Reply:
> In the rest of the logic I assume that the internet was present before turning it off.

You mean it was present for the previous part of the test? If yes, I think we should check it before executing that part, not after.

Anyway, not a merge blocker.